src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -1796,12 +1796,8 @@ public void reset() {
 	@Override
 	public void calculate(CompilationUnit sourceCompilationUnit, List<CtType<?>> types) {
 		this.sourceCompilationUnit = sourceCompilationUnit;
-		// reset the import scanner between each compilationunit
-		if (env.isAutoImports()) {
-			this.importsContext = new ImportScannerImpl();
-		} else {
-			this.importsContext = new MinimalImportScanner();
-		}
+		// warning: the import scanner used in this pretty-printer has been set once and for all in the constructor
+		// this is required to reuse import when multiple classes are in the same compilation unit
```

Comment:
> this is required to reuse import when multiple classes are in the same compilation unit

If they are in the same compulation unit, then we enter only once in this method and it's already using the same importContext. 
However it could mess things with different compilation units: let's say we are in no-autoimport and we have two CU, the first one use B but have a collision, then it should import B, if the second CU also use B, even without collision it will be imported too and that's not the expected behaviour.